exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/JoinUtils.java

```diff
@@ -203,35 +203,27 @@ public static void addLeastRestrictiveCasts(LogicalExpression[] leftExpressions,
   }
 
   /**
-   * Utility method to check if a subquery (represented by its root RelNode) is provably scalar. Currently
-   * only aggregates with no group-by are considered scalar. In the future, this method should be generalized
-   * to include more cases and reconciled with Calcite's notion of scalar.
+   * Utility method to check if a subquery (represented by its root RelNode) is provably scalar.
    * @param root The root RelNode to be examined
    * @return True if the root rel or its descendant is scalar, False otherwise
    */
   public static boolean isScalarSubquery(RelNode root) {
-    DrillAggregateRel agg = null;
-    RelNode currentrel = root;
-    while (agg == null && currentrel != null) {
-      if (currentrel instanceof DrillAggregateRel) {
-        agg = (DrillAggregateRel)currentrel;
-      } else if (currentrel instanceof RelSubset) {
-        currentrel = ((RelSubset)currentrel).getBest() ;
-      } else if (currentrel.getInputs().size() == 1) {
-        // If the rel is not an aggregate or RelSubset, but is a single-input rel (could be Project,
-        // Filter, Sort etc.), check its input
-        currentrel = currentrel.getInput(0);
-      } else {
-        break;
-      }
-    }
-
-    if (agg != null) {
-      if (agg.getGroupSet().isEmpty()) {
-        return true;
+    RelMetadataQuery relMetadataQuery = RelMetadataQuery.instance();
+    RelNode currentRel = root;
+    for (; ; ) {
```

Comment:
If the currentRel happens to be null we should not enter the loop.  Better to keep the while(currentRel != null) and have the return FALSE at the end.